hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/common/CachingBlockManager.java

```diff
@@ -433,13 +434,13 @@ private void addToCacheAndRelease(BufferData data, Future<Void> blockFuture) {
     }
 
     try {
-      Await.result(blockFuture);
+      blockFuture.get(); //TODO consider calling get(long timeout, TimeUnit unit) instead
```

Comment:
if there's a todo item here, it'll need fixing or at least understanding of why a fix isn't needed

Reply:
would it be possible to agree a timeout value - maybe 1 hour? - otherwise, this could block indefinitely

Reply:
this is also the same problem that would occur with the previous Await.result call